tests/foreman/ui/test_settings.py

```diff
@@ -353,9 +353,43 @@ def test_positive_update_email_delivery_method_sendmail():
     :CaseImportance: Critical
 
     :CaseLevel: Acceptance
-
-    :CaseAutomation: notautomated
     """
+    mail_config_default_param = {
+        "delivery_method": "",
+        "email_reply_address": "",
+        "email_subject_prefix": "",
+        "sendmail_arguments": "",
+        "sendmail_location": "",
+        "send_welcome_email": "",
+    }
+    mail_config_default_param = {
+        content: entities.Setting().search(query={'search': f'name={content}'})[0]
+        for content in mail_config_default_param
+    }
+    mail_config_new_params = {
+        "delivery_method": "Sendmail",
+        "email_reply_address": f"root@{ssh.settings.server.hostname}",
+        "email_subject_prefix": [gen_string('alpha')],
+        "sendmail_location": "/usr/sbin/sendmail",
+        "send_welcome_email": "Yes",
+    }
+
+    with session:
+        try:
+            for mail_content, mail_content_value in mail_config_new_params.items():
+                session.settings.update(mail_content, mail_content_value)
+            assert session.settings.send_test_mail()[0] == "Email was sent successfully"
+            assert (
+                ssh.command(
+                    "grep {} /var/mail/root".format(mail_config_new_params["email_subject_prefix"])
+                ).return_code
+                == 0
+            )
```

Comment:
Not blocking, but this hanging comparison makes me uncomfortable. Perhaps break it down to multiple lines:
```python
            cmd = "grep {} /var/mail/root".format(mail_config_new_params["email_subject_prefix"])
            assert ssh.command(cmd).return_code == 0
```
In fact, this way it takes fewer lines of code than your current approach.